staging/src/k8s.io/apiserver/pkg/storage/value/encrypt/envelope/kmsv2/cache.go

```diff
@@ -65,14 +68,18 @@ func (c *simpleCache) get(key []byte) value.Read {
 }
 
 // set caches the record for the key
-func (c *simpleCache) set(key []byte, transformer value.Read) {
+func (c *simpleCache) set(key []byte, transformer value.Read, providerName string) {
```

Comment:
Since it must not change during the life of the cache, I would make `providerName` a field on the cache and take it as input to `newSimpleCache`.